docs/bigquery-usage.rst

```diff
@@ -227,6 +227,20 @@ Update all writable metadata for a table
    ...     SchemaField(name='age', type='int', mode='required)]
    >>> table.update()  # API request
 
+Get rows from a table's data:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> dataset = client.dataset('dataset_name')
+   >>> table = dataset.table(name='person_ages')
+   >>> rows, next_page_token = table.data(max_results=100)  # API request
+   >>> rows.csv.headers
```

Comment:
I mean to refer to the [GoF flyweight pattern](https://en.wikipedia.org/wiki/Flyweight_pattern).  I imagined that `rows.csv` would be a stateless, iterator-like object which would be useful for generating a CSV output stream from the extracted data.